jsonnet/exclude-rules.libsonnet

```diff
@@ -0,0 +1,95 @@
+local excludeRule(rule, excludedRules) =
+  if std.length(excludedRules) == 0 then
+    [rule]
+  else if (('alert' in rule && 'alert' in excludedRules[0]) && rule.alert == excludedRules[0].alert) ||
+          (('record' in rule && 'record' in excludedRules[0]) && rule.record == excludedRules[0].record) then
+    []
+  else
+    [] + excludeRule(rule, excludedRules[1:]);
+
+local excludeRuleGroup(group, excludedGroups) =
+  if std.length(excludedGroups) == 0 then
+    [group.rules]
+  else if (group.name == excludedGroups[0].name) then
+    [excludeRule(rule, excludedGroups[0].rules) for rule in group.rules]
+  else
+    [] + excludeRuleGroup(group, excludedGroups[1:]);
+
+{
+  excludeRules(o): {
+    local exclude(o) = o {
+      [if (o.kind == 'PrometheusRule') then 'spec']+: {
+        groups: std.filterMap(
+          function(group) !std.member(excludedRuleGroups, group.name),
+          function(group)
+            group {
+              rules: std.flattenArrays(
+                excludeRuleGroup(group, excludedRules.groups)
+              ),
+            },
+          super.groups,
+        ),
+      },
+    },
+    [k]: exclude(o[k])
+    for k in std.objectFields(o)
+  },
+
+  local excludedRuleGroups = [
+    'kube-apiserver-availability.rules',
+  ],
+
+  local excludedRules = {
```

Comment:
Wdyt about converting this data structure to an array and reducing nesting?

Reply:
good idea :+1: